src/translator/cache.cpp

```diff
@@ -0,0 +1,165 @@
+#include "cache.h"
+
+#include <cstdlib>
+
+#include "translation_model.h"
+
+namespace marian {
+namespace bergamot {
+
+size_t TranslationCache::hash(const TranslationCache::CacheKey &key) {
+  size_t seed = 42;
+  for (auto &word : key.words) {
+    size_t hashWord = static_cast<size_t>(word.toWordIndex());
+    util::hash_combine<size_t>(seed, hashWord);
+  }
+
+  util::hash_combine<size_t>(seed, key.model.modelId());
+  return seed;
+}
+
+ThreadSafeL4Cache::ThreadSafeL4Cache(const ThreadSafeL4Cache::Config &config)
+    : epochManagerConfig_(config.ebrQueueSize, std::chrono::milliseconds(config.ebrIntervalInMilliseconds),
+                          config.ebrNumQueues),
+      cacheConfig_(config.sizeInMB * 1024 * 1024, std::chrono::seconds(config.timeToLiveInMilliseconds),
+                   config.removeExpired),
+      service_(epochManagerConfig_),
+      context_(service_.GetContext()) {
+  // There is only a single cache we use. However, L4 construction API given by example demands we give it a string
+  // identifier.
+  const std::string cacheIdentifier = "global-cache";
+  hashTableIndex_ = service_.AddHashTable(L4::HashTableConfig(
+      cacheIdentifier, L4::HashTableConfig::Setting{static_cast<uint32_t>(config.numBuckets)}, cacheConfig_));
+}
```

Comment:
Umm.. on the `hashTableIndex_` thread.
```
hashTableIndex_ = service.AddHashTable(...) // On construction
```

The above variable is used everywhere else. I think it maybe possible for me to `const` this through some construction ordering. `cacheIdentifier` can be constexpr. 

`hashTableIndex` *will* come from L4, obtained value used ahead. i.e, I'm not doing `constexpr` on `hashTableIndex_`being 0`. 

Does this work for you? @XapaJIaMnu